syntax = "proto3";

option java_multiple_files = true;
option objc_class_prefix = "HLW";

service MatrixCalc {
  rpc BlockAdd (Request) returns (Reply) {}
  rpc BlockMult (Request) returns (Reply) {}
}

message Request {
  Mat matA = 1;
  Mat matB = 2;
}

message Reply {
  Mat matResult = 1;
}

message Mat {
  repeated Vector row = 1;
}

message Vector {
  repeated int32 value = 1;
}
